content/backend/graphql-js/7-subscriptions.md

```diff
@@ -1,81 +1,132 @@
+Subscriptions are a GraphQL feature that allows a server to
+send data to its clients when a specific _event_ happens.
+Subscriptions are usually implemented with
+[WebSockets](https://en.wikipedia.org/wiki/WebSocket). In
+that setup, the server maintains a steady connection to its
+subscribed client. This also breaks the
+"Request-Response-Cycle" that were used for all previous
+interactions with the API.
+
+Instead, the client initially opens up a long-lived
+connection to the server by sending a _subscription query_
+that specifies which _event_ it is interested in. Every time
+this particular event happens, the server uses the
+connection to push the event data to the subscribed
+client(s).
 
 ### Implementing GraphQL subscriptions
 
-We will be using `PubSub` from the `graphql-yoga` library that we have already been using for our GraphQL server to implement subscriptions to the following _events_:
+We will be using `PubSub` from the `apollo-server` library
+that we have already been using for our GraphQL server to
+implement subscriptions to the following _events_:
 
 - A new model is **created**
 - An existing model **updated**
 - An existing model is **deleted**
 
-You will do this by first adding an instance of `PubSub` to the context, just as we did with `PrismaClient`, and then calling its methods in the resolvers that handle each of the above events.
+You will do this by first adding an instance of `PubSub` to
+the context, just as we did with `PrismaClient`, and then
+calling its methods in the resolvers that handle each of the
+above events.
 
 ### Setting up `PubSub`
 
 <Instruction>
 
-Open your `index.js` file where we instantiate the server and add the following code:
+Open your `index.js` file where we instantiate the server
+and add the following code:
 
 ```graphql(path=".../hackernews-node/src/index.js)
 // ... previous import statements
-const { PubSub } = require('graphql-yoga')
+const { PubSub } = require('spollo-server')
 
 const pubsub = new PubSub()
 ```
 
 </Instruction>
 
-Here, you're creating an instance of `PubSub` and storing it in the variable `pubsub`, just as we stored an instance of `PrismaClient` in the variable `prisma`.
+Here, you're creating an instance of `PubSub` and storing it
+in the variable `pubsub`, just as we stored an instance of
+`PrismaClient` in the variable `prisma`.
 
 <Instruction>
 
-Now, in the same file, add `pubsub` to the context, just as did with `prisma`:
+Now, in the same file, add `pubsub` to the context, just as
+did with `prisma`:
 
-```graphql(path=".../hackernews-node/src/index.js)
-const server = new GraphQLServer({
-  typeDefs: './src/schema.graphql',
+```{11}graphql(path=".../hackernews-node/src/index.js)
```

Comment:
Let's include the file name here for the diff to be shown on the file🙂
```suggestion
```js{11}graphql(path=".../hackernews-node/src/index.js)
```